File: js/src/gdb/tests/enum-printers.py
# Test that we can find pretty-printers for enums.
# flake8: noqa: F821

import mozilla.prettyprinters


@mozilla.prettyprinters.pretty_printer("unscoped_no_storage")
class my_typedef(object):
    def __init__(self, value, cache):
        pass

    def to_string(self):
        return "unscoped_no_storage::success"


@mozilla.prettyprinters.pretty_printer("unscoped_with_storage")
class my_typedef(object):
    def __init__(self, value, cache):
        pass

    def to_string(self):
        return "unscoped_with_storage::success"


@mozilla.prettyprinters.pretty_printer("scoped_no_storage")
class my_typedef(object):
    def __init__(self, value, cache):
        pass

    def to_string(self):
        return "scoped_no_storage::success"


@mozilla.prettyprinters.pretty_printer("scoped_with_storage")
class my_typedef(object):
    def __init__(self, value, cache):
        pass

    def to_string(self):
        return "scoped_with_storage::success"


run_fragment("enum_printers.one")
assert_pretty("i1", "unscoped_no_storage::success")
assert_pretty("i2", "unscoped_with_storage::success")
assert_pretty("i3", "scoped_no_storage::success")
assert_pretty("i4", "scoped_with_storage::success")

File: js/src/gdb/tests/enum-printers.cpp
#include "gdb-tests.h"

#include <stdint.h>

enum unscoped_no_storage { EnumValue1 };

enum unscoped_with_storage : uint8_t { EnumValue2 };

enum class scoped_no_storage { EnumValue3 };

enum class scoped_with_storage : uint8_t { EnumValue4 };

FRAGMENT(enum_printers, one) {
  unscoped_no_storage i1 = EnumValue1;
  unscoped_with_storage i2 = EnumValue2;
  scoped_no_storage i3 = scoped_no_storage::EnumValue3;
  scoped_with_storage i4 = scoped_with_storage::EnumValue4;

  breakpoint();

  use(i1);
  use(i2);
  use(i3);
  use(i4);
}
